_posts/alex_lepsa/2018-03-22-Spring-Method-Security-Using-JWTs.md

```diff
@@ -0,0 +1,342 @@
+---
+layout: post
+title:  "Using JWT's with Spring Security's @PreAuthorize annotation for method specific security"
+breadcrumb: true
+author: alex_lepsa
+date: 2018-03-22
+categories: alex_lepsa
+tags:
+    - Spring Security
+    - JWT
+teaser:
+    info: We explore how to implement a spring security strategy that statelessly authorizes a user using via JWT, allowing for method level permissions using the @PreAuthorize annotation.
+    image: alex_lepsa/2018-03/spring_jwt_thumb.jpg
+header:
+    version: small
+    title: Software Engineering Blog
+    image: header-logo-crop.png
+    icon: icon-blog 
+---
+
+Much has been written about integrating JWT's into spring security, and in fact Pivotal has included more and more first-class support for JWT's in their recent releases. That said, one thing that seems to be missing is a summary on how to stitch JWT's into an existing application using the `@PreAuthorize` annotation for fine-grain access control. We will be referencing our [SONG](https://www.overture.bio/song) application from [Overture](https://www.overture.bio/).
+
+# Enter SONG, the hero we need #
+
+SONG is an open source system for validating and tracking meta-data about raw data submissions, assigning identifiers to entities of interest, and managing the state of the raw data with regards to publication and access. This is an existing, in production, application that we want to update to authorize users in a stateless manner using JWT's. In our case we already have a JWT provider in our [EGO](https://github.com/overture-stack/ego) service which takes care of authenticating our user and then passes us a JWT we can then use to verify the users' permissions.
```

Comment:
I would break this paragraph in two and rephrase a bit the second portion.

"Used in production, we wanted to update it to authorize users in a stateless manner using JWT". EGO, our JWT provider takes cares of ...."

Reply:
I've reworded this a bit bit not exactly like above, it needed some attention so thanks for this as well!